spack/package.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Arbor(CMakePackage, CudaPackage):
+    """Arbor is a high-performance library for computational neuroscience
+    simulations."""
+
+    homepage = 'https://github.com/arbor-sim/arbor/'
+    git      = 'https://github.com/arbor-sim/arbor.git'
+    url      = 'https://arbor.readthedocs.io/en/latest/'
+    maintainers = ['bcumming', 'halfflat']
+
+    version('master', branch='master', submodules=True)
+    version('0.5', 'd0c8a4c7f97565d7c30493c66249be794d1dc424de266fc79cecbbf0e313df59', url='https://github.com/arbor-sim/arbor/releases/download/v0.5/arbor-v0.5-full.tar.gz')
+
+    variant('assertions', default=False, description='Enable arb_assert() assertions in code.')
+    variant('doc', default=False, description='Build documentation.')
+    variant('mpi', default=False, description='Enable MPI support')
+    variant('neuroml', default=True, description='Build NeuroML support library.')
+    variant('python', default=True, description='Enable Python frontend support')
+    variant('vectorize', default=False, description='Enable vectorization of computational kernels')
+
+    # https://arbor.readthedocs.io/en/latest/install/build_install.html?highlight=requirements#compilers
+    conflicts('%gcc@:8.3')
+    conflicts('%clang@:8.0')
+    # Cray compiler v9.2 and later is Clang-based.
+    conflicts('%cce@:9.1')
+    conflicts('%intel')
+
+    depends_on('cmake@3.12:', type='build')
+
+    # misc dependencies
+    depends_on('nlohmann-json')
+    depends_on('cuda@10:', when='+cuda')
+    depends_on('libxml2', when='+neuroml')
+
+    # mpi
+    depends_on('mpi', when='+mpi')
+    depends_on('py-mpi4py', when='+mpi+python', type=('build', 'run'))
```

Comment:
mpi4py is not a compile type dep, and only a runtime dep if the user wants to configure their own communicator.

Reply:
This is something we discussed when writing the spack package.
Because we have spack handling the dependencies, it is easy to ask it to configure mpi4py if installing arbor with mpi and python.